internal/tiltfile/encoding/yaml.go

```diff
@@ -50,29 +68,55 @@ func decodeYAML(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tup
 		return nil, fmt.Errorf("%s arg must be a string or blob. got %s", fn.Name(), contents.Type())
 	}
 
-	return yamlStringToStarlark(s, "")
+	return yamlStreamToStarlark(s, "")
 }
 
-func yamlStringToStarlark(s string, source string) (starlark.Value, error) {
-	var decodedYAML interface{}
-	err := yaml.Unmarshal([]byte(s), &decodedYAML)
+func decodeYAML(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	v, err := decodeYAMLStream(thread, fn, args, kwargs)
 	if err != nil {
-		errmsg := "error parsing YAML"
-		if source != "" {
-			errmsg += fmt.Sprintf(" from %s", source)
-		}
-		return nil, errors.Wrap(err, errmsg)
+		return nil, err
+	}
+	l, ok := v.(*starlark.List)
+	if !ok {
+		return nil, fmt.Errorf("internal error: expected decodeYAMLStream to return a %T, but it returned a %T", starlark.List{}, l)
+	}
+	if l.Len() != 1 {
+		return nil, fmt.Errorf("expected a yaml document but found a yaml stream (documents separated by `---`). use %s instead to decode a yaml stream", decodeYAMLStreamN)
```

Comment:
what happens if you call decode_yaml('') on an empty string? does it hit this error case?

Reply:
good catch